### ISSUE TITLE
The rar files are not extracted

### ISSUE THREAD

**Issue by alon24** [levic92/LCExtractor#14]
I installed the plugin, on windows, with deluge running as a service
I enabled the plugin
I installed 7zip
It does not unrar the torrent after finish.

Am I missing a step here?

**Comment by levic92** [comment]
The instructions are in the readme. Make sure you are using Python 2.7 or you will need to build your own egg file or rename it to match your installed version. All that needs to happen is install the plugin and then configure the options you want. There are no other steps. From my experience the Deluge GUI has bugs in it and I have had more success with the webui. I do not use Windows but I know others have used this successfully on windows.

**Comment by deafeningquiet** [comment]
I was having similar issues and finally managed to figure it out.

I'll post what managed to fix for me in case it helps anyone else.

I started by running:
deluged-debug.exe -L debug -l "R:\Temp\deluge.log"

where I found an error similar to the following when the torrent finished downloading:
`eventmanager:59 Event handler TorrentFinishedEvent failed in <bound method Core._on_torrent_finished of <lcextractor.core.Core object at 0x063DCC70>> with exception (2, 'CreateProcess', 'The system cannot find the file specified.')`

A bit of digging led me to this old forum post:
http://forum.deluge-torrent.org/viewtopic.php?f=12&t=52295&start=10

Which doesn't have a lot of help, but one final suggestion.
So I uninstalled 7zip x64 and reinstalled the x86 build with nothing else to try

Stopped and restarted deluged in debug mode again and got a different error:
`The process cannot access the file because it is being used by another process.`

Getting closer!

A quick Google search turned up:
https://www.reddit.com/r/torrents/comments/3zee2t/deluge_error_file_in_use_by_another_process/

which mentioned the Extractor plugin (which LCExtractor is based from) had conflicts with AV installed.

In my case, I made an exclusion in Windows Defender for my Downloads directory for torrents and everything magically started working. The torrent was successfully extracted.

TLDR:
1) Install x86 version of 7zip (I'm currently at v18.01)
2) Make an exclusion for your torrents download directory in whatever AV program you may run.

I'm not thrilled about needing to whitelist my torrents directory from AV though...

Anyways, hope this helps someone out there.

**Comment by nckchris** [comment]
I seem to be having a similar issue, but I'm not sure. It could very well be user error as this is a totally new setup for me.

I'm using Sonarr/Radarr on Windows 7 to import torrents into Deluge. When the torrents complete they don't seem to be extracting automatically all of the time. The odd part is, I've had a couple extract properly and then Sonarr/Radarr did their thing and moved everything where they were supposed to. In the last few days the extraction seems to have stopped working and I can't seem to figure out why. If I manually extract the files with 7zip Sonarr/Radarr pick up on it and automate everything. I've just enabled logging for Deluge, so that may shed some light on things, but I was curious if anyone had any suggestions. I've been back through everything, and nothing jumps out, but I very well could have missed something.

Do the files have to seed to the set ratio before LCExtractor does its thing?

Thanks!

**Comment by DemonCow5761** [comment]
I've been having things randomly not extract for quite some time now.  Never really checked logs before as I didn't know how and I'm at my PC a lot and I run this on my normal rig so not too big of a deal to extract a few now and again by hand.  I just whitelisted my folder with the x64 version to see if it works.  If it doesn't I'll try the x86 next.  Maybe check out those logs more if neither work I guess.  I'll try n report back if anything works.

@nckchris 
No they normally just extract right away when it's done.  I assume if you had it working and it still sporadically working your settings are right.  You can check your lcextract.conf in your appdata/roaming/deluge and make sure things are marked true still as the main deluge app (opposed to web app) is kinda buggy n sometimes shows your settings all unticked.

**Comment by DemonCow5761** [comment]
Didn't work for me :-(

**Comment by DemonCow5761** [comment]
Checked logs with deluged-debug.exe -L debug -l "D:\Library\deluge.log" nothing showed up about lcextractor or the torrents failing to extract. /shrug

**Comment by levic92** [comment]
I could only reliably get the settings to stick when using the deluge web UI (windows GUI program did not work). If you didn't see any logs then that most likely means the plugin isn't running. Make sure you are using the correct python version that corresponds with the version in the egg file.

**Comment by DemonCow5761** [comment]
I dunno I don't even change anything and it'll work fine and randomly 1 or 2 wont work and then it works fine.  When i go into the config file everything is on and fine in there and I don't even touch anything in the client settings for a long time only webui or the config file.  I feel like I even saw the 7zip window open for a second on a somewhat larger file where it should stay around longer and then go away when it didn't work.  It only happens like once or twice a week out of like 10-20 rar files and then sometimes it works fine for weeks with no issue.

**Comment by DemonCow5761** [comment]
Well it has been working for me for 3 weeks without a single file not getting extracted.  I deleted the lcextractor.conf~ file in appdata (which I assume is just some temporary config file?)  It was the exact same as my lcextractor.conf file but it seems to have fixed my problem I guess.
